src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1413,39 +1413,26 @@ public override BoundNode VisitAnonymousObjectCreationExpression(BoundAnonymousO
         {
             Debug.Assert(!IsConditionalState);
 
-            int receiverSlot = -1;
-            var arguments = node.Arguments;
-            var constructor = node.Constructor;
-            for (int i = 0; i < arguments.Length; i++)
-            {
-                var argument = arguments[i];
-                TypeSymbolWithAnnotations argumentType = VisitRvalueWithResult(argument);
-                var parameter = constructor.Parameters[i];
-                ReportArgumentWarnings(argument, argumentType, parameter);
+            var anonymousType = (NamedTypeSymbol)node.Type;
+            Debug.Assert(anonymousType.IsAnonymousType);
 
-                // https://github.com/dotnet/roslyn/issues/24018 node.Declarations includes
-                // explicitly-named properties only. For now, skip expressions
-                // with implicit names. See NullableReferenceTypesTests.AnonymousTypes_05.
-                if (node.Declarations.Length < arguments.Length)
-                {
-                    continue;
-                }
+            var arguments = node.Arguments;
+            var argumentTypes = arguments.SelectAsArray((arg, visitor) => visitor.VisitRvalueWithResult(arg), this);
```

Comment:
>VisitRvalueWithResult [](start = 82, length = 21)

Do we need to ensure that the argument types are speakable? Or can we assert that they are?
For instance, you have an unconstrained `T`, but you null-test it, then use it to construct an anonymous type. Just like in method type inference, we should infer speakable types.